tests/test_sysadmin.py

```diff
@@ -0,0 +1,387 @@
+from openedx.core.djangolib.markup import Text
+
+TEST_MONGODB_LOG = {
+    "host": MONGO_HOST,
+    "port": MONGO_PORT_NUM,
+    "user": "",
+    "password": "",
+    "db": "test_xlog",
+}
+
+
+class SysadminBaseTestCase(SharedModuleStoreTestCase):
+    """
+    Base class with common methods used in XML and Mongo tests
+    """
+
+    TEST_REPO = "https://github.com/edx/edx4edx_lite.git"
+    TEST_BRANCH = "testing_do_not_delete"
+    TEST_BRANCH_COURSE = CourseLocator.from_string(
+        "course-v1:MITx+edx4edx_branch+edx4edx"
+    )
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    def setUp(self):
+        """Setup test case by adding primary user."""
+        super().setUp()
+        self.user = UserFactory.create(
+            username="test_user", email="test_user+sysadmin@edx.org", password="foo"
+        )
+        self.client = Client()
+
+    def _setstaff_login(self):
+        """Makes the test user staff and logs them in"""
+        GlobalStaff().add_users(self.user)
+        self.client.login(username=self.user.username, password="foo")
+
+    def _add_edx4edx(self, branch=None):
+        """Adds the edx4edx sample course"""
+        post_dict = {
+            "repo_location": self.TEST_REPO,
+            "action": "add_course",
+        }
+        if branch:
+            post_dict["repo_branch"] = branch
+        return self.client.post(reverse("sysadmin_courses"), post_dict)
+
+    def _rm_edx4edx(self):
+        """Deletes the sample course from the XML store"""
+        def_ms = modulestore()
+        course_path = "{}/edx4edx_lite".format(os.path.abspath(settings.DATA_DIR))
+        try:
+            # using XML store
+            course = def_ms.courses.get(course_path, None)
+        except AttributeError:
+            # Using mongo store
+            course = def_ms.get_course(CourseLocator("MITx", "edx4edx", "edx4edx"))
+
+        # Delete git loaded course
+        response = self.client.post(
+            reverse("sysadmin_courses"),
+            {
+                "course_id": str(course.id),
+                "action": "del_course",
+            },
+        )
+        self.addCleanup(self._rm_glob, f"{course_path}_deleted_*")
+
+        return response
+
+    def _rm_glob(self, path):
+        """
+        Create a shell expansion of passed in parameter and iteratively
+        remove them.  Must only expand to directories.
+        """
+        for path in glob.glob(
+            path
+        ):  # lint-amnesty, pylint: disable=redefined-argument-from-local
+            shutil.rmtree(path)
+
+    def _mkdir(self, path):
+        """
+        Create directory and add the cleanup for it.
+        """
+        os.mkdir(path)
+        self.addCleanup(shutil.rmtree, path)
+
+
+@override_settings(
+    MONGODB_LOG=TEST_MONGODB_LOG,
+    GIT_REPO_DIR=settings.TEST_ROOT / f"course_repos_{uuid4().hex}",
+)
+@unittest.skipUnless(
+    settings.FEATURES.get("ENABLE_SYSADMIN_DASHBOARD"),
```

Comment:
@arslanashraf7 we have removed this feature flag `ENABLE_SYSADMIN_DASHBOARD` so it needs to be removed from tests too.

Reply:
Updated.